File: AuthenticationDemo/Database/Entites/Hero.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace AuthenticationDemoAPI.Database.Entites;

public class Hero
{
    [Key]
    public int Id { get; set; }

    [Column(TypeName = "nvarchar(32)")]
    public string HeroName { get; set; } = string.Empty;

    [Column(TypeName = "nvarchar(32)")]
    public string RealName { get; set; } = string.Empty;

    [Column(TypeName = "nvarchar(32)")]
    public string Place { get; set; } = string.Empty;

    [Column(TypeName = "smallint")]
    public short DebutYear { get; set; } = 0;
}

File: AuthenticationDemoClient/src/app/_models/hero.ts
export interface Hero {
  id: number;
  heroName: string;
  realName: string;
  place: string;
  debutYear: number;
}

export function resetHero(): Hero {
  return { id: 0, heroName: '', realName: '', place: '', debutYear: 0 };
}
